core/common/src/main/java/alluxio/grpc/GrpcChannelKey.java

```diff
@@ -0,0 +1,314 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.grpc;
+
+import alluxio.collections.Pair;
+import alluxio.conf.AlluxioConfiguration;
+
+import com.google.common.base.MoreObjects;
+import io.netty.channel.Channel;
+import io.netty.channel.EventLoopGroup;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
```

Comment:
what do all these new annotations do? The old code did not have them.

Reply:
It's a new annotation defined below. It's easier to just tag fields to be included in identity calculation, rather than changing `hashCode` and `equals` implementations all the time. There has been some bugs in the past due to tediousness of prior way of maintaining the identity of channel.